src/sql/workbench/contrib/notebook/browser/markdownWYSIWYGLinkHander.ts

```diff
@@ -0,0 +1,91 @@
+import { URI } from 'vs/base/common/uri';
+import * as path from 'vs/base/common/path';
+
+export class MarkdownWYSIWYGLinkHandler {
+	private _isAnchorLink: boolean | undefined;
+	private _isFile: boolean | undefined;
+	private _isAbsolutePath: boolean | undefined;
+	private _notebookDirectory: string;
+	private _href: URI | undefined;
+
+	constructor(private _notebookURI: URI, private _path: string | HTMLAnchorElement) {
+		if (typeof this._path === 'string') {
+			this._href = URI.parse(this._path);
+			this._isFile = this._href.scheme === 'file';
+			this._isAbsolutePath = path.isAbsolute(this._path);
+			this._isAnchorLink = this._path.includes('#') && this._isFile;
+		} else {
+			this._href = this._path.attributes['href']?.nodeValue ? URI.parse(this._path.attributes['href']?.nodeValue) : undefined;
+			this._isFile = this._path.protocol === 'file:';
+			this._isAnchorLink = this._href?.fragment ? true : false;
+			this._isAbsolutePath = this._path.attributes['is-absolute']?.nodeValue === 'true' ? true : false;
+		}
+		this._notebookDirectory = this._notebookURI ? path.dirname(this._notebookURI.fsPath) : '';
+	}
+
+	public getLinkUrl(): string {
+		switch (typeof this._path) {
+			case 'string': {
+				if (this._isFile && !this._isAbsolutePath && !this._isAnchorLink) {
+					const relativePath = (this._path).replace(/\\/g, path.posix.sep);
+					const linkUrl = path.resolve(this._notebookDirectory, relativePath);
+					return linkUrl;
+				}
+				return this._path;
+			}
+			case 'object': {
+				if (this._href) {
```

Comment:
Why all these truthy checks? Are there values you expect to be empty/undefined sometimes?